### PR TITLE
add invert output option

### LINKED ISSUE #408: Add: Inverted Output
can we add an option to the "Switch" mode for "Inverted Relay" (maybe only for the Shelly1, as that has a Dry Contact Relay and can be used for various things other than a power relay)

Description:
So the HomeKit icon shows as ON when the relay is OPEN (OFF), and HomeKit icon shows as OFF when relay is CLOSED (ON).

as per disscussion here https://github.com/mongoose-os-apps/shelly-homekit/issues/406

### PR DESCRIPTION
**Pull request by timoschilling**
closes #408

### CODE REVIEW

**Comment by andyblac** [comment]
another one need a conflict correction 😄

File: mos.yml [hunk 1]
```diff
@@ -40,6 +40,7 @@ config_schema:
   - ["sw", "o", {title: "Switch settings", abstract: true}]
   - ["sw.name", "s", "", {title: "Name of the switch"}]
   - ["sw.enable", "b", true, {title: "Enable this switch in the accessory"}]
+  - ["sw.inverted", "b", true, {title: "Invert output, set to true for normally open output"}]
```
Comment (rojer) [inline mos.yml:43]: we have `in_inverted`, let's call this option `out_inverted`

**Review by rojer — COMMENTED** [review]
overall looks good but there is one more thing - when constructing components, we need to reset output to non-inverted as a common baseline, so for example if you change switch into window covering, that know nothing about inverting outputs, outputs do not remain inverted.
it's not ideal, and represents a leak in abstractions, so if you have better idea how to manage it - welcome.
but for now, we'll need something similar to [this](https://github.com/mongoose-os-apps/shelly-homekit/blob/c335332eb96c05d84411d9a67d7929cabafd620c/src/shelly_main.cpp#L318-L320).

**Comment by timoschilling** [comment]
Maybe I miss something, but I think we don't need to reset it. `inverted` is a config property of `sw`, which means that if the user switches from `sw` to `window` it reboots the devise which means a new output is created and which will use the `wc` setting and will never be in `inverted` mode.

**Comment by rojer** [comment]
> it reboots the devise

except it doesn't. `*restart_required = true` does not mean reboot, it means HAP server restart.
HAP server gets stopped, components get destroyed and re-created. however, peripherals - the Inputs, Outputs, temp sensors, etc, do not. so if inverted setting would be preserved and new components created with inverted output(s).

**Comment by timoschilling** [comment]
review applyed

**Comment by rojer** [comment]
i don't see changes to shelly_main

**Comment by timoschilling** [comment]
thanks for the hint, lost during a rebase

**Comment by rojer** [comment]
merged, thanks!

**Pull request merged**